python/doc/developer_guide/architecture.rst

```diff
@@ -184,65 +184,67 @@ Dependencies
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Special functions (optional)          | `Boost <http://www.boost.org/>`_                                                | 1.46              |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `NLopt <http://ab-initio.mit.edu/nlopt>`_                                       | 2.4               |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `CMinpack <http://devernay.free.fr/hacks/cminpack/>`_                           | 1.3               |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `Ceres Solver <http://ceres-solver.org>`_                                       | 1.11              |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `Dlib <http://dlib.net/>`_                                                      | 18.18             |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `Bonmin <https://projects.coin-or.org/Bonmin>`_                                 | 1.8.7             |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| Optimization (optional)               | `Ipopt <https://coin-or.github.io/Ipopt/>`_                                     | 3.11.9            |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| CSV parser (optional)                 | flex                                                                            | 2.5.33            |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| CSV parser (optional)                 | bison                                                                           | 2.4               |
++---------------------------------------+---------------------------------------------------------------------------------+-------------------+
+| XML support (optional)                | `LibXml2 <http://xmlsoft.org/>`_                                                | 2.6.27            |
```

Comment:
beware I just merged hdf5 adding a line here support so you might get a conflict